org.eclipse.lsp4e/src/org/eclipse/lsp4e/internal/FutureUtil.java

```diff
@@ -20,25 +22,20 @@
 public class FutureUtil {
 
 	/**
-	 * Combines two async lists of results into a single list by adding all the
-	 * elements of the second list to the first one.
+	 * Combines two async lists of results into a new single list containing all
+	 * elements of the first and the second result.
 	 *
-	 * @param <T>
-	 *            Result type
-	 * @param accumulator
-	 *            One async result
-	 * @param another
-	 *            Another async result
 	 * @return Async combined result
 	 */
-	public static <T> CompletableFuture<List<T>> addAll(CompletableFuture<List<T>> accumulator,
-			CompletableFuture<List<T>> another) {
-		CompletableFuture<List<T>> res = accumulator.thenCombine(another, (a, b) -> {
-			a.addAll(b);
-			return a;
+	public static <T> CompletableFuture<List<T>> join(final CompletableFuture<List<T>> firstResult,
+			final CompletableFuture<List<T>> secondResult) {
+		final CompletableFuture<List<T>> result = firstResult.thenCombine(secondResult, (firstList, secondList) -> {
+			final List<T> combinedList = new ArrayList<>(firstList);
```

Comment:
should we not create the list with the size firstList.size() + 1 to avoid a potential copy of the internal array?